crates/bevy_ecs/src/system/query.rs

```diff
@@ -50,330 +57,424 @@ use core::{
+/// (`Option<&ComponentA>`) because they broaden the amount of entities kinds that can be accessed.
+/// This is especially true of a query that _only_ fetches optional components, as the query would
+/// iterate over all entities in the world.
 ///
-/// Optional components increase the number of entities a query has to match against.
-/// This can hurt iteration performance, especially if the query solely consists of only optional components, since the query would iterate over each entity in the world.
+/// There are two types of [component storage types]: [`Table`] and [`SparseSet`]. [`Table`] offers
+/// fast iteration speeds, but slower insertion and removal speeds. [`SparseSet`] is the opposite:
+/// it offers fast component insertion and removal speeds, but slower iteration speeds.
 ///
-/// The following table compares the computational complexity of the various methods and operations, where:
+/// The following table compares the computational complexity of the various methods and
+/// operations, where:
 ///
-/// - **n** is the number of entities that match the query,
-/// - **r** is the number of elements in a combination,
-/// - **k** is the number of involved entities in the operation,
-/// - **a** is the number of archetypes in the world,
-/// - **C** is the [binomial coefficient], used to count combinations.
-///   <sub>n</sub>C<sub>r</sub> is read as "*n* choose *r*" and is equivalent to the number of distinct unordered subsets of *r* elements that can be taken from a set of *n* elements.
+/// - **n** is the number of entities that match the query.
+/// - **r** is the number of elements in a combination.
+/// - **k** is the number of involved entities in the operation.
+/// - **a** is the number of archetypes in the world.
+/// - **C** is the [binomial coefficient], used to count combinations. <sub>n</sub>C<sub>r</sub> is
+///   read as "*n* choose *r*" and is equivalent to the number of distinct unordered subsets of *r*
+///   elements that can be taken from a set of *n* elements.
 ///
 /// |Query operation|Computational complexity|
-/// |:---:|:---:|
-/// |[`iter`]\[[`_mut`][`iter_mut`]]|O(n)|
-/// |[[`iter().for_each()`][`for_each`]\[[`iter_mut().for_each()`][`for_each`]],<br>[`par_iter`]\[[`_mut`][`par_iter_mut`]]|O(n)|
-/// |[`iter_many`]\[[`_mut`][`iter_many_mut`]]|O(k)|
-/// |[`iter_combinations`]\[[`_mut`][`iter_combinations_mut`]]|O(<sub>n</sub>C<sub>r</sub>)|
-/// |[`get`]\[[`_mut`][`get_mut`]]|O(1)|
-/// |([`get_`][`get_many`])[`many`]|O(k)|
-/// |([`get_`][`get_many_mut`])[`many_mut`]|O(k<sup>2</sup>)|
-/// |[`single`]\[[`_mut`][`single_mut`]],<br>[`single`]\[[`_mut`][`single_mut`]]|O(a)|
-/// |Archetype based filtering ([`With`], [`Without`], [`Or`])|O(a)|
+/// |-|-|
+/// |[`iter`]\[[`_mut`][`iter_mut`]\]|O(n)|
+/// |[`iter().for_each()`][`for_each`],<br />[`iter_mut().for_each()`][`for_each`],<br/>[`par_iter`]\[[`_mut`][`par_iter_mut`]\]|O(n)|
```

Comment:
It might make sense to do the same `[_mut]` syntax for the `for_each` row.  

```suggestion
/// |[`iter\[_mut\]().for_each()`][`for_each`],<br/>[`par_iter`]\[[`_mut`][`par_iter_mut`]\]|O(n)|
```